client/components/editor/teaching-elements/Assessment/TextSelection/Selector.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <div
+    @mouseup="onMouseup">
+    <div
+      v-html="structure"
+      :class="{ editing: this.isEditing }"
+      class="selector ql-editor"/>
+  </div>
+</template>
+
+<script>
+import {
+  generateStructure,
+  getSelectedWords,
+  isBlast,
+  isSelected,
+  nodeMapper,
+  modifySelectionNodes
+} from './helpers';
+import isEmpty from 'lodash/isEmpty';
+import Selections from './Selections';
+import curry from 'lodash/curry';
+
+export default {
+  props: {
+    correct: { type: Array, required: true },
+    isEditing: { type: Boolean, required: true },
+    content: { type: String, required: true }
+  },
+  data() {
+    const selections = new Selections(this.correct);
+    const processNode = selections.processNode.bind(selections);
+    const mapper = curry(nodeMapper)(processNode);
+    return {
+      selections,
+      structure: generateStructure(this.content, mapper)
```

Comment:
Why this isn't implemented as computed property because at the end of day it is _computed_ from content 😉

Reply:
Whenever you can avoid currying by using lambda please do it so:
```js
generateStructure(this.content, (node, i) => nodeMapper(processNode, node, i));
```

Reply:
This isn't a computed property because it should be computed only once. It has a dependency on `this.selections` and that would mean regenerating and re-rendering  whole component each time a selection is made or removed. It's much better to manipulate few specific DOM nodes and add/remove classes on them.

Reply:
> It has a dependency on this.selections and that would mean regenerating and re-rendering whole component each time a selection is made or removed.

Through bound `processNode` though, I'm not 100% sure that would force Vue into tracking `selections` and recalculate it on change? 🤔